### PR TITLE
feat: send invoice email notifications

### LINKED ISSUE #14: Notify user about new invoice
We want to send an email as soon as a new invoice is charged to the customer. 

It should include a link to download the invoice.

Needs #54

### LINKED ISSUE #54: Add language to customer

### PR DESCRIPTION
**Pull request by anbraten**
closes #54 

closes #14

### CODE REVIEW

File: packages/server/src/mail.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+};
+
+const templates: Record<string, Template> = {
+  newInvoice: {
+    subject: {
+      en: handlebars.compile(`{{ project.name }} - New invoice {{ invoice.number }}`),
+      de: handlebars.compile(`{{ project.name }} - Neue Rechnung {{ invoice.number }}`),
+    },
+    text: {
+      en: handlebars.compile(`
```
Comment (mariusheine) [inline packages/server/src/mail.ts:51]: @anbraten For typesafety we should do
```suggestion
      en: handlebars.compile<{ customer: Customer; invoice: Invoice }>(`
```

here and also at the de template

File: packages/server/src/mail.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+type Template = {
+  subject: Record<Languages, ReturnType<typeof handlebars.compile>>;
+  text: Record<Languages, ReturnType<typeof handlebars.compile>>;
+};
+
+const templates: Record<string, Template> = {
+  newInvoice: {
+    subject: {
+      en: handlebars.compile(`{{ project.name }} - New invoice {{ invoice.number }}`),
+      de: handlebars.compile(`{{ project.name }} - Neue Rechnung {{ invoice.number }}`),
```
Comment (mariusheine) [inline packages/server/src/mail.ts:48]: @anbraten 

```suggestion
      en: handlebars.compile<{ project: Project; invoice: Invoice }>(`{{ project.name }} - New invoice {{ invoice.number }}`),
      de: handlebars.compile<{ project: Project; invoice: Invoice }>(`{{ project.name }} - Neue Rechnung {{ invoice.number }}`),
```

**Review by mariusheine — CHANGES_REQUESTED** [review]

File: packages/server/src/mail.ts (hunk 1 above)
Comment (anbraten) [reply packages/server/src/mail.ts:51]: Wasn't aware that's possible. Nice one.

Comment (anbraten) [reply packages/server/src/mail.ts:51]: @mariusheine Tried that, but it then complains that the type of the Record is not correct. So we either need to remove the Record type or ignore the generic.

Comment (mariusheine) [reply packages/server/src/mail.ts]: @anbraten I played a bit around and came up with a bit complex but quite well working type. (or a few more types :rofl:) What do you think about it?

Comment (mariusheine) [reply packages/server/src/mail.ts]: @anbraten 33ac5d7ee2745f76e88f6facbc7b17d08588e096

Comment (mariusheine) [reply packages/server/src/mail.ts]: @anbraten Ok I had to fix some checks but now it is totally fine, but they are two commits now ;)

Comment (anbraten) [reply packages/server/src/mail.ts]: @mariusheine Quite some code for a "simple" type, but fine for me.

**Review by mariusheine — APPROVED** [review]

**Pull request merged**